### PR TITLE
Refactor proposition classes and DTOs

### PR DESCRIPTION
**Pull request by corecl4ud**
# Main Changes:
- Renamed "PropositionCriterion" to "EventCriterion" to better align with the new data structure where criteria now belong to events rather than challenges.

# Updated Class Structures:

- In the EventModel, we've added a list of PropositionModel to hold all propositions associated with an event, and a list of EventCriterionModel to store all criteria associated with an event.

- The PropositionModel no longer contains a list of EventCriteria since all event criteria are active in all propositions.

- The CriterionModel now has a list of EventCriterionModel to store the criteria associated with an event.

- In the GroupRatingModel, it was added a reference to the "EventCriterion" to represent the criterion associated with a group rating.

# Migration of PropositionCriteria:
- Migrated "PropositionCriteria" to "EventCriteria" in the DTO "PropositionDTO" to reflect the change in terminology and data structure.

# Controller Updates:
- The PropositionController has been updated to match the changes in the service layer and data structure, ensuring that the API endpoints correctly handle the updated class structures.

### CODE REVIEW

File: hackweek-backend/Models/CriterionModel.cs
Comment (helenaluz) [inline hackweek-backend/Models/CriterionModel.cs:1]: Ve isso aqui!! Non-nullable property 'EventCriteria' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
 Check warning on line 10 in hackweek-backend/Models/CriterionModel.cs

GitHub Actions
/ build-macOS-latest
Non-nullable property 'EventCriteria' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.
    }	    }
}	}

File: hackweek-backend/Models/EventCriterionModel.cs
Comment (helenaluz) [inline hackweek-backend/Models/EventCriterionModel.cs:1]: Non-nullable property 'Event' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.

 Check warning on line 9 in hackweek-backend/Models/EventCriterionModel.cs

GitHub Actions
/ build-macOS-latest

Non-nullable property 'Event' must contain a non-null value when exiting constructor. Consider declaring the property as nullable.


        public CriterionModel? Criterion { get; set; }

    }

}

File: hackweek-backend/Models/EventModel.cs
Comment (helenaluz) [inline hackweek-backend/Models/EventModel.cs:1]: erro git hub

File: hackweek-backend/Services/CriterionService.cs
Comment (helenaluz) [inline hackweek-backend/Services/CriterionService.cs:1]: erro github

File: hackweek-backend/Services/RatingService.cs
Comment (helenaluz) [inline hackweek-backend/Services/RatingService.cs:1]: erro git hub

**Review by helenaluz — APPROVED** [review]

**Review by helenaluz — APPROVED** [review]

**Review by raphael-teodoro — APPROVED** [review]

**Pull request merged**